pkg/cloud/azure/actuators/machine_scope.go

```diff
@@ -159,22 +160,22 @@ func (m *MachineScope) storeMachineStatus(machine *machinev1.Machine) (*machinev
 	return m.MachineClient.UpdateStatus(machine)
 }
 
-// Close the MachineScope by updating the machine spec, machine status.
-func (m *MachineScope) Close() {
+// Store the machine spec and machine status.
+func (m *MachineScope) Store() error {
```

Comment:
Is it always the case that you want both `Spec` and `Status` updated? Are there any cases where you would want or need to update `Status` only?

Reply:
This is done in one go atm, let's keep the scope of the PR narrowed and do any farther in follow ups.